com.unity.render-pipelines.high-definition/Runtime/Compositor/AlphaInjection.cs

```diff
@@ -0,0 +1,61 @@
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.Rendering.HighDefinition;
+using System;
+
+namespace UnityEngine.Rendering.HighDefinition.Compositor
+{
+    [Serializable, VolumeComponentMenu("Post-processing/Custom/AlphaInjection")]
+    internal sealed class AlphaInjection : CustomPostProcessVolumeComponent, IPostProcessComponent
+    {
+        internal class ShaderIDs
+        {
+            public static readonly int k_AlphaTexture = Shader.PropertyToID("_AlphaTexture");
+            public static readonly int k_InputTexture = Shader.PropertyToID("_InputTexture");
+        }
+
+        Material m_Material;
+
+        public bool IsActive() => m_Material != null;
+
+        public override CustomPostProcessInjectionPoint injectionPoint => CustomPostProcessInjectionPoint.BeforePostProcess;
+
+        public override void Setup()
+        {
+            if (Shader.Find("Hidden/Shader/AlphaInjection") != null)
+                m_Material = new Material(Shader.Find("Hidden/Shader/AlphaInjection"));
```

Comment:
Regarding the comment from the other review.

The shader could still live in Compositor folders, the only difference is that instead of doing Shader.Find you get it from the resources. 

It's fine to do the ^ I guess, is just that is fairly different than the usual pattern.

Reply:
hmm.. good catch, I haven't think about this one.
One of the issue here is that if it is not part of resources asset, it will not be embed in player. So this isn't a good idea. (only referenced shader are embed). So you need to add it here. 

In the past torbjorn have done work to allow to extent our resources thing: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/4454 - guess we need to resurect this PR and use it for those shaders above. Let's merge this PR as it is and we will discuss about the extension.

Reply:
Ok, Seb explained to me the issue. I will switch to the "usual pattern" before we merge.